app/particle/particle_simulation.py

```diff
@@ -168,13 +205,32 @@ def change_num_active_workers():
                 new_distribution[new_thread].add(particle)
 
         for i in range(1, params.num_threads):
-            print("sending "+str(i))
+#            print("sending "+str(i))
             params.partitions[i].particles = new_distribution[i]
             params.comm.send(new_distribution[i], dest = i, tag = 11)
     else:
-        print("receiving "+str(params.rank))
+#        print("receiving "+str(params.rank))
         params.partitions[params.rank].receive_new_particles()
-        #print("received "+str(params.rank))
+#        print("received "+str(params.rank))
+
+def change_num_particles():
+#    util.debug("change_num_particles is called")
+    if params.num_particles == params.new_num_particles:
+        pass
+    elif params.num_particles < params.new_num_particles:
+        if params.rank is 0:
+            for i in range(params.num_particles, params.new_num_particles):
+                add_one_particle(i)
+            for i in range(1, params.num_threads):
+#                print("sending " + str(i))
+                params.comm.send(params.partitions[i].particles, dest = i, tag = 11)
+        else:
+            params.partitions[params.rank].receive_new_particles()
+    else:
+        # this is the case when params.num_particles > params.new_num_particles
+        # note that params.new_num_particles == params.num_particles - 1
```

Comment:
Is this always guaranteed to be the case?  i.e. is only one particle allowed to be removed at a time?